crates/runtime/src/dataconnector.rs

```diff
@@ -393,35 +392,31 @@ impl<T: ListingTableConnector + Display> DataConnector for T {
 
         let url = self
             .get_object_store_url(dataset)
```

Comment:
`get_object_store_url` needs to be changed to return correct `DataConnectorError`. Returning wrapped errors is last resort if we can't detect it.